File: WePeiYang/lostfound/src/main/java/com/twtstudio/tjwhm/lostfound/search/SearchContract.java
package com.twtstudio.tjwhm.lostfound.search;

import com.twtstudio.tjwhm.lostfound.base.BaseContract;
import com.twtstudio.tjwhm.lostfound.waterfall.WaterfallBean;

/**
 * Created by tjwhm on 2017/7/6.
 **/

public interface SearchContract {
    public interface SearchUIView extends BaseContract.BaseView {
        void setSearchData(WaterfallBean waterfallBean);
    }

    public interface SearchPresenter extends BaseContract.BasePresenter {
        void loadSearchData(String keyword,int page);
        void setSearchData(WaterfallBean waterfallBean);
    }
}

File: WePeiYang/lostfound2/src/main/java/com/example/lostfond2/search/SearchContract.kt
package com.example.lostfond2.search

import com.example.lostfond2.service.MyListDataOrSearchBean

interface SearchContract {
    interface SearchUIView {
        fun setSearchData(waterfallBean: MutableList<MyListDataOrSearchBean>)
    }

    interface SearchPresenter {
        fun loadSearchData(keyword: String, page: Int)
        fun setSearchData(waterfallBean: MutableList<MyListDataOrSearchBean>)
    }

}